### ISSUE TITLE
Pathway description save fails if text includes certain characters

### ISSUE THREAD

**Issue by khanspers** [wikipathways/wikipathways.org#46]
Saving of pathway description (on the pathway page) fails completely, and without warning, if the text includes certain characters, in this case ampersand. The outcome if that nothing is saved and the description is simply blank. 

For example, saving a description with this URL doesn't work:
http://www.genome.jp/kegg-bin/show_pathway?org_name=hsa&mapno=04014

While this version of the URL for the same pathway works just fine:
http://www.genome.jp/dbget-bin/www_bget?map04014

Testing the ampersand by itself on WP4 it seems like the culprit. Kept the title of this vague since it may not be limited to this one character.

**Comment by Artoria2e5** [comment]
Feeding `foobar&amp;` in gives the normal `foobar&`, so it's probably PageEditor itself trying to interpret entities and being too strict with it. Clicking "edit" button shows `foobar&` as source, suggesting that PageEditor fails to re-escape the content too.

**Comment by Artoria2e5** [comment]
The broken saving part is here:
https://github.com/wikipathways/wikipathways.org/blob/92e2bb99b3e564e25ba13f557d631e7e5459ca34/wpi/extensions/PageEditor/PageEditor.php#L78

Setting the `nodeValue` is nearly equivalent to just putting a lump of text without escaping right between the opening and closing tags. The XML routines will then validate the content and declare it invalid.

One replacement of `nodeValue` is `textcontent`, which probably will escape everything potentially problematic on save. That of course also gets rid of "good" or well-formed entities and tags for styling. Alternatively some sort of intermediate HTML parser can be used for error tolerance: just tell it to parse and then somehow graft the well-formed tree onto the `$description` node.

**Comment by DeniseSl22** [comment]
Yes, found this for several PWs as well, either from a too long URL or a "weird" symbol (which can even be an ' in names of chemicals)... Perhaps we should write up a page explaining how to write a description (or incorporate what follows next with already existing texts), which symbols are not allowed (if we can not fix it right away), and add a "ethical writing" paragraph (example: http://proteopedia.org/wiki/index.php/Proteopedia:Guidelines_for_Ethical_Writing).
